pkg/controller/managedresources/controller.go

```diff
@@ -140,13 +143,29 @@ func (r *Reconciler) reconcile(mr *resourcesv1alpha1.ManagedResource, log logr.L
 			return reconcile.Result{}, err
 		}
 
-		for _, value := range secret.Data {
+		for key, value := range secret.Data {
 			var (
 				decoder    = yaml.NewYAMLOrJSONDecoder(bytes.NewReader(value), 1024)
 				decodedObj map[string]interface{}
 			)
 
-			for err := decoder.Decode(&decodedObj); err == nil; err = decoder.Decode(&decodedObj) {
+			for i := 0; true; i++ {
+				err := decoder.Decode(&decodedObj)
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					decodingError := &decodingError{
+						err:               err,
+						secret:            fmt.Sprintf("%s/%s", secret.Name, secret.Namespace),
```

Comment:
```suggestion
						secret:            fmt.Sprintf("%s/%s", secret.Namespace, secret.Name),
```
Typically, it is in this order

Reply:
Corrected, PTAL :)